src/domain/transaction/midnight/midnight_transaction.rs

```diff
@@ -130,6 +138,205 @@ where
         // 3. Construct the full transaction
         // 4. Update transaction data with proof request IDs
 
+        let wallet_seed = self.signer.wallet_seed();
+
+        let indexer_client = self.provider.get_indexer_client();
+
+        // TODO: We should probably move this up one level
+        // This still requires `MIDNIGHT_LEDGER_TEST_STATIC_DIR` environment variable to be set (limitation by LedgerContext test resolver)
+        // We should check with the Midnight team if we can use a different constructor for LedgerContext
+        let mut sync_manager = SyncManager::<QuickSyncStrategy>::new(
+            indexer_client,
+            wallet_seed,
+            to_midnight_network_id(&self.network.network),
+        )
+        .map_err(|e| TransactionError::UnexpectedError(e.to_string()))?;
+
+        sync_manager
```

Comment:
Do we have somewhere comment what sync is doing?

It would be great that this complex flow is somehow documented.